src/components/Courses/Courses.jsx

```diff
@@ -33,30 +41,80 @@ function Courses(props) {
 		return resAuthors;
 	}
 
+	function onSearchCourse(text) {
+		let resArray = [];
+
+		if (text === '') {
+			setCourses(allCoursesItem);
+		} else {
+			if (courses) {
+				courses?.forEach((element) => {
+					const foundTitle = element.title.toLowerCase().indexOf(text, 0);
+					const foundId = element.id.toLowerCase().indexOf(text, 0);
+					if (foundTitle > -1 || foundId > -1) {
+						resArray.push(element);
+					}
+				});
+			}
+
+			if (resArray.length > 0) {
+				setCourses(resArray);
+			}
+		}
+	}
+
+	useEffect(() => {
+		//const currentToken = localStorage.getItem('courseUserToken');
+		//if (currentToken === null) {
+		if (isLogIn === false) {
```

Comment:
this kind of logic is usually abstracted away into separate components, here's an example:

App.jsx
```
<Routes>
				<Route path='/' element={<AuthorizedGuard><Login /></AuthorizedGuard>} />
```

AuthorizedGuard
```
// useEffect(isLogIn) from here
```